### PR TITLE
[web] Cleanup touch and mouse event adapters

### PR DESCRIPTION
**Pull request by mdebbar**
This PR cleans up a big source of complexity in our pointer handling code. It also drops support for Safari versions prior to 13 ([caniuse](https://caniuse.com/mdn-api_pointerevent)).

Right now, we throw a hard error when running in a browser that doesn't support `PointerEvent`s. Should we turn it into a soft warning, and just disable gesture/pointer handling on those browsers?

For users who need to support older versions of Safari, they can try this polyfill: https://github.com/wessberg/pointer-events (thanks @ditman!)

Fixes https://github.com/flutter/flutter/issues/116141

### CODE REVIEW

File: lib/web_ui/test/engine/semantics/semantics_helper_test.dart [hunk 1]
```diff
@@ -122,14 +119,7 @@ void testMain() {
 
       test('Non-relevant events should be forwarded to the framework',
           () async {
-        DomEvent event;
-        if (_defaultSupportDetector.hasPointerEvents) {
-          event = createDomPointerEvent('pointermove');
-        } else if (_defaultSupportDetector.hasTouchEvents) {
-          event = createDomTouchEvent('touchcancel');
```
Comment (ditman) [inline lib/web_ui/test/engine/semantics/semantics_helper_test.dart:129]: `createDomTouchEvent` and `createDomMouseEvent` are now probably unused in `dom.dart`?

**Review by ditman — COMMENTED** [review]
From what I can find, this drops support for: iPhone 6 and lower (5S, etc...) (those are stuck forever in iOS 12).

The oldest model that we'd support would be: iPhone 6S (iOS 15.7), released in September, 2018.

Should we deploy an app with this change applied so we can take a look at it with a bunch of devices?

File: lib/web_ui/lib/src/engine/pointer_binding.dart [hunk 2]
```diff
@@ -80,9 +80,11 @@ class SafariPointerEventWorkaround {
 }
 
 class PointerBinding {
-  PointerBinding(this.flutterViewElement, this._keyboardConverter)
-    : _pointerDataConverter = PointerDataConverter(),
-      _detector = const PointerSupportDetector() {
+  PointerBinding(
+    this.flutterViewElement,
+    this._keyboardConverter, [
+    this._detector = const PointerSupportDetector(),
```
Comment (ditman) [inline lib/web_ui/lib/src/engine/pointer_binding.dart:86]: I don't think we need this `PointerSupportDetector` anymore?

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/pointer_binding.dart:86]: This was (and still is) useful for testing

**Comment by mdebbar** [comment]
> From what I can find, this drops support for: iPhone 6 and lower (5S, etc...) (those are stuck forever in iOS 12).
> 
> The oldest model that we'd support would be: iPhone 6S (iOS 15.7), released in September, 2018.

Bummer 😞 maybe we should wait a little longer?

File: lib/web_ui/test/engine/semantics/semantics_helper_test.dart (hunk 1 above)
Comment (mdebbar) [reply lib/web_ui/test/engine/semantics/semantics_helper_test.dart:129]: `createDomTouchEvent` yes, but `createDomMouseEvent` is used in other places.

**Review by yjbanov — APPROVED** [review]
LGTM if LGT @ditman

**Comment by ditman** [comment]
> Bummer 😞 maybe we should wait a little longer?

I'm gathering some data about usage, maybe we can extract some guideline to know *when* something can be deprecated :P

**Review by ditman — APPROVED** [review]
(Approving because I didn't have other than nits)

**Pull request merged**